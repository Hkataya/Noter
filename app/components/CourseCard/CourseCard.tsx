import React from 'react';
import styled from 'styled-components';
import Button from '../Button/Button';
import { CourseType } from '../../reducers/entities/types';
import { CourseActionCreatorType } from '../../actions/courses';

type BgProps = {
  thumbnail: string;
};
const Background = styled.div.attrs({
  className: 'flex overflow-hidden bg-cover bg-center'
})<BgProps>`
  background-image: url(${p => p.thumbnail});
`;

const LeftItem = styled.div.attrs({
  className: 'w-1/3'
})``;

const RightItem = styled.div.attrs({
  className: 'w-2/3 p-4 bg-white rounded-lg'
})`
  background: rgb(28, 26, 46);
  background: linear-gradient(
    120deg,
    rgba(28, 26, 46, 0.8999230033810399) 0%,
    rgba(28, 26, 46, 0.8999230033810399) 100%
  );
`;

const TitleWrapper = styled.div.attrs({
  className: 'flex justify-around  text-white'
})``;
const Title = styled.h1.attrs({
  className: 'text-white font-bold text-lg text-justify mb-4'
})``;

const Description = styled.p.attrs({
  className: 'mt-4 text-gray-600 text-sm text-justify'
})``;

const ListItemWrapper = styled.div.attrs({
  className: 'mt-4 text-gray-600 text-sm mt-4 text-justify'
})``;

const ListItem = styled.li.attrs({
  className: 'mt-1'
})``;

type Props = CourseType &
  CourseActionCreatorType & {
    directToCoursePage: () => void;
  };

const CourseCard = (props: Props) => {
  const {
    id,
    title,
    chapters,
    videos,
    thumbnail,
    removeCourse,
    directToCoursePage
  } = props;
  return (
    <div className="w-108 shadow-lg rounded-lg overflow-hidden m-5">
      <Background thumbnail={thumbnail || ''}>
        <LeftItem />
        <RightItem>
          <TitleWrapper>
            <Title>{title}</Title>
            <button
              onClick={() => {
                if (removeCourse) removeCourse(id);
              }}
              type="button"
              className="ml-10"
            >
              <i className="fas fa-trash" />
            </button>
          </TitleWrapper>
          <hr />
          <Description>description</Description>
          <ListItemWrapper>
            <ul>
              <ListItem>
                nb of chapters:
                {chapters ? chapters.length : 0}
              </ListItem>
              <ListItem>
                nb of videos:
                {videos ? videos.length : 0}
              </ListItem>
              <ListItem>duration: </ListItem>
            </ul>
          </ListItemWrapper>
          <Button handleButtonClick={directToCoursePage}> Watch </Button>
        </RightItem>
      </Background>
      <div className="shadow w-full  rounded-lg bg-grey-light">
        <div
          className="bg-purple-900 py-1 text-center"
          style={{ width: '45%' }}
        />
      </div>
    </div>
  );
};

export default CourseCard;
