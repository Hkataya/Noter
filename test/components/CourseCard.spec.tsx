import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import CourseCard from '../../app/components/CourseCard/CourseCard';

test('loads and displays course card', async () => {
  const { asFragment } = render(
    <CourseCard
      id="123"
      title="test"
      removeCourse={() => {}}
      directToCoursePage={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('clicks direct to course page button', async () => {
  const directToCoursePage = jest.fn();

  render(
    <CourseCard
      id="123"
      title="test"
      description="description"
      removeCourse={() => {}}
      directToCoursePage={directToCoursePage}
    />
  );

  fireEvent.click(screen.getByText('Watch'));
  expect(directToCoursePage).toHaveBeenCalled();
});
