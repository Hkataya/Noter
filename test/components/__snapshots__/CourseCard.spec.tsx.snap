// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays course card 1`] = `
<DocumentFragment>
  <div
    class="w-96 shadow-lg rounded-t-lg m-5"
  >
    <div
      class="sc-AxhUy dBRLnL flex  bg-cover bg-center rounded-t-lg"
    >
      <div
        class="sc-AxgMl fFPQpl w-1/3"
      />
      <div
        class="sc-AxheI bMWDco w-2/3 p-5 bg-white rounded-lg"
      >
        <div
          class="flex justify-end -mt-4"
        >
          <div
            class="relative inline-block"
          >
            <div>
              <button
                class="sc-AxjAm StDqN inline-flex justify-center w-full rounded-md  text-sm leading-5 font-medium text-gray-700 hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue active:bg-gray-50 active:text-gray-800 transition ease-in-out duration-150"
                role="button"
                type="button"
              >
                <i
                  class="fas fa-ellipsis-v"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-Axmtr hBVTto flex text-white mb-4"
        >
          <h1
            class="sc-AxmLO cAKCgh text-white font-bold text-lg text-justify break-all"
          >
            test
          </h1>
        </div>
        <hr />
        <p
          class="sc-fzozJi jxgYyc mt-4 text-gray-600 text-sm text-justify break-all"
        >
          description
        </p>
        <div
          class="sc-fzoLsD fqEnxe mt-4 text-gray-600 text-sm mt-4 text-justify"
        >
          <ul>
            <li
              class="sc-fzpans kkFAbe mt-1"
            >
              <i
                class="fas fa-stream mr-3"
              />
              Sections:   0
            </li>
            <li
              class="sc-fzpans kkFAbe mt-1"
            >
              <i
                class="fas fa-clock mr-3"
              />
              duration:  
            </li>
          </ul>
        </div>
        <div
          class="flex justify-end mt-3"
        >
          <button
            class="sc-AxhCb ccMvAG px-3 py-2 bg-purple-900 text-white text-xs font-bold uppercase rounded"
            type="button"
          >
            Watch
          </button>
        </div>
      </div>
    </div>
    <div
      class="shadow w-full rounded-b-lg bg-grey-light"
    >
      <div
        class="bg-purple-900 py-1 text-center rounded-b-lg"
        style="width: 45%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
